src/components/autocomplete.js

```diff
@@ -0,0 +1,201 @@
+            filteredSuggestions: [],
+            // Whether or not the suggestion list is shown
+            showSuggestions: false,
+            // What the user has entered
+            userInput: "",
+        };
+    }
+
+    // Event fired when the input value is changed
+    onChange = e => {
+        const { suggestions } = this.props;
+        const userInput = e.currentTarget.value;
+
+        // Filter our suggestions that don't contain the user's input
+        const filteredSuggestions = suggestions.filter(
+            suggestion =>
+                suggestion.toLowerCase().indexOf(userInput.toLowerCase()) > -1
+        );
+
+        // Update the user input and filtered suggestions, reset the active
+        // suggestion and make sure the suggestions are shown
+        this.setState({
+            activeSuggestion: 0,
+            filteredSuggestions,
+            showSuggestions: true,
+            userInput: e.currentTarget.value
+        });
+    };
+
+    // Event fired when the user presses a key down
+    onKeyDown = e => {
+        const { activeSuggestion, filteredSuggestions } = this.state;
+        const spacesData = this.props.data;
+        // const history = this.props.history;
+
+        // User pressed the enter key, update the input and close the
+        // suggestions
+        if (e.keyCode === 13) {
+            let title = filteredSuggestions[activeSuggestion];
+            let space = spacesData.filter(obj => {
+                return obj.title === title
+            })[0];
+
+            if (space) {
+                this.props.history.push({
+                    pathname: '/view-space',
+                    state: { data: space }
+                })
+            }
+        }
+    };
+
+    render() {
+        const {
+            onChange,
+            onKeyDown,
+            state: {
+                activeSuggestion,
+                filteredSuggestions,
+                showSuggestions,
+                userInput
+            }
+        } = this;
+
+        let suggestionsListComponent;
+        const data = this.props.data;
+
+        if (showSuggestions && userInput) {
+            if (filteredSuggestions.length) {
+                suggestionsListComponent = (
+                    <ul className="suggestions">
+                        {filteredSuggestions.map((suggestion, index) => {
+                            let className;
+
+                            // Flag the active suggestion with a class
+                            if (index === activeSuggestion) {
+                                className = "suggestion-active";
+                            }
+
+                            let link_location = data.filter(obj => {
+                                return obj.title === suggestion
+                            })[0];
+
+                            return (
+                                <Link to={{ state: { data: link_location }, pathname: "/view-space" }}
```

Comment:
I'm assuming you made it so that the search doesn't display a list of "search results" and instead goes directly to the view-space page?

Reply:
IMO it makes sense that if a user selects a specific entry it would take them to that entry. For enter-key though we can change the behavior once search-results is more polished.